File: src/cli/worker.ts
import { pool } from "../db/pool.js";
import { startWorker } from "../queue/worker.js";

const { stop } = await startWorker();

const shutdown = async (sig: string) => {
  console.log(`[worker-cli] ${sig} received, stopping`);
  await stop();
  await pool.end();
  process.exit(0);
};
process.on("SIGINT", () => shutdown("SIGINT"));
process.on("SIGTERM", () => shutdown("SIGTERM"));

File: src/parts_research/cli/worker.py
"""Entry point воркера Этапа 2:

    python -m parts_research.cli.worker

Запускает run_forever() с graceful-shutdown на SIGINT/SIGTERM (drain inflight).
Можно поднимать несколько процессов на одну общую очередь."""

from __future__ import annotations

import asyncio

from ..queue.worker import run_forever


def main() -> None:
    asyncio.run(run_forever())


if __name__ == "__main__":
    main()
